### PR TITLE
fix: be defensive around use of window.WebSocket

### PR DESCRIPTION
**Pull request by keithamus**
This fixes #19, #23 by being more defensive around the use of `window.WebSocket`.

It also provides more informational errors when attempting to call `.send()` on a WebSocket that is not `OPEN`.

### CODE REVIEW

File: robust-websocket.js [hunk 1]
```diff
@@ -75,6 +77,12 @@
     }
 
     self.send = function() {
+      if (realWs.readyState !== self.OPEN) {
+        throw Object.assign(
+          new Error('An attempt was made to use a WebSocket that is not usable'),
+          { name: 'InvalidStateError' }
+        )
+      }
```
Comment (nathanboktae) [inline robust-websocket.js:85]: Why would we try to re-implement the WebSocket standard? calling `realWs.send.apply(...)` should just throw this error.

File: robust-websocket.js [hunk 2]
```diff
@@ -11,6 +11,8 @@
   }
 })(function(global, navigator) {
 
+  var WebSocket = global.WebSocket
```
Comment (nathanboktae) [inline robust-websocket.js:14]: This doesn't make sense. either a) `WebSocket` is native and supported and never replaced, or b) it is polyfilled and we shouldn't close over and force the users to understand ordering of this library and the polyfill.

File: robust-websocket.js [hunk 3]
```diff
@@ -239,5 +247,16 @@
...
+  ['CONNECTING' , 'OPEN', 'CLOSING', 'CLOSED'].forEach(function (name) {
+    const descriptor = {
+      writable: false,
+      enumerable: true,
+      configurable: false,
+      value: WebSocket[name],
+    }
+    Object.defineProperty(RobustWebSocket, name, descriptor)
+    Object.defineProperty(RobustWebSocket.prototype, name, descriptor)
+  })
```
Comment (nathanboktae) [inline robust-websocket.js:259]: I'd be glad to take this change at `robust-websocket`'s new home, [nathanboktae/robust-websocket](https://github.com/nathanboktae/robust-websocket)

**Review by nathanboktae — CHANGES_REQUESTED** [review]

File: robust-websocket.js (hunk 1 above)
Comment (keithamus) [reply robust-websocket.js:85]: As previously stated - `realWs.send` is not always present

File: robust-websocket.js (hunk 2 above)
Comment (keithamus) [reply robust-websocket.js:14]: The reason for this change is for another option: WebSocket has been overriden by browser extensions and users want to ensure that RobustWebSocket's implementation doesn't change from under them.

File: robust-websocket.js (hunk 1 above)
Comment (nathanboktae) [reply robust-websocket.js:85]: But that's not what this code is checking. If you want to check for th3 existence of send and throw an Error, that is acceptible.

**Comment by keithamus** [comment]
Closing this one in favour of https://github.com/nathanboktae/robust-websocket/pull/2

**Pull request closed without merging**